### PR TITLE
Add Preset fields for controlling when a logical hint should be placed.

### PR DESCRIPTION
**Pull request by henriquegemignani**

### CODE REVIEW

File: randovania/layout/preset_migration.py [hunk 1]
```diff
@@ -584,7 +584,10 @@ def _migrate_v25(preset: dict) -> dict:
 
     return preset
 
+
 def _migrate_v26(preset: dict) -> dict:
+    preset["configuration"]["minimum_available_locations_for_hint_placement"] = 0
+    preset["configuration"]["minimum_location_weight_for_hint_placement"] = 0.0
```
Comment (duncathan) [inline randovania/layout/preset_migration.py:590]: this is bad form and you know it :P

File: randovania/generator/filler/player_state.py [hunk 2]
```diff
@@ -219,6 +219,26 @@ def filter_usable_locations(self, locations_weighted: WeightedLocations) -> Weig
 
         return locations_weighted
 
+    def should_have_hint(self, pickup: PickupEntry, current_uncollected: UncollectedState,
+                         all_locations_weighted: WeightedLocations) -> bool:
+
+        if not pickup.item_category.is_major:
```
Comment (duncathan) [inline randovania/generator/filler/player_state.py:225]: shouldn't this be checking the interesting pickup thing?

**Review by duncathan — CHANGES_REQUESTED** [review]

File: randovania/layout/preset_migration.py (hunk 1 above)
Comment (henriquegemignani) [reply randovania/layout/preset_migration.py:590]: The reusing a preset version? I don't care, that was just merged lol

The bunch of similar fields? I could add a class but also bleh...

File: randovania/generator/filler/player_state.py (hunk 2 above)
Comment (henriquegemignani) [reply randovania/generator/filler/player_state.py:225]: If it was picked at all it's an interesting pickup.

File: randovania/layout/preset_migration.py (hunk 1 above)
Comment (duncathan) [reply randovania/layout/preset_migration.py:590]: my presets are ruined ;~;

**Pull request merged**